src/main/java/taxi/service/CarServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package taxi.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import taxi.dao.CarDao;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Car;
+import taxi.model.Driver;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private CarDao carDao;
+
+    @Override
+    public Car create(Car car) {
+        return carDao.create(car);
+    }
+
+    @Override
+    public Car get(Long id) {
+        return carDao.get(id).orElseThrow(() ->
+                new RuntimeException("Id don't found " + id));
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return carDao.getAll();
+    }
+
+    @Override
+    public Car update(Car car) {
+        return carDao.update(car);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        if (!get(car.getId()).getDrivers().contains(driver)) {
```

Comment:
we don't need this check, db will handle this situation and throw an exception if driver doesn't exist
and don't forget to call  the update method after

Reply:
done